code/modules/power/apc/apc_main.dm

```diff
@@ -463,10 +463,11 @@
 			update()
 		if("emergency_lighting")
 			emergency_lights = !emergency_lights
-			for(var/obj/machinery/light/L in area)
-				if(!initial(L.no_low_power)) //If there was an override set on creation, keep that override
-					L.no_low_power = emergency_lights
-					INVOKE_ASYNC(L, TYPE_PROC_REF(/obj/machinery/light/, update), FALSE)
+			for(var/turf/area_turf as anything in area.get_contained_turfs())
+				for(var/obj/machinery/light/area_light in area_turf)
+					if(!initial(area_light.no_low_power)) //If there was an override set on creation, keep that override
+						area_light.no_low_power = emergency_lights
+						INVOKE_ASYNC(area_light, TYPE_PROC_REF(/obj/machinery/light/, update), FALSE)
```

Comment:
applies to a lot of these actually

Reply:
It pretty close but it seems more common machines (APCs, lights) are slower

Light switches(above)

APCs
![image](https://github.com/tgstation/tgstation/assets/51863163/c93f9be8-d249-453f-9ec5-388128e20f0a)

Lights 
![image](https://github.com/tgstation/tgstation/assets/51863163/e8c032e6-6ef1-449d-a81e-1638adc96c1c)